analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,401 @@
+samples_per_celltype_df <- epic_df |>
+  dplyr::filter(!is_other) |>
+  dplyr::mutate(celltype_present = fraction > 1e-4) |>
+  dplyr::group_by(epic_celltype, project_diagnosis) |>
+  dplyr::summarize(n_samples = sum(celltype_present)) 
+
+# convert number to proportion
+samples_per_celltype_df <- epic_df |>
+  dplyr::filter(is_other) |>
+  dplyr::count(project_diagnosis) |>
+  dplyr::right_join(samples_per_celltype_df) |>
+  dplyr::mutate(prop_samples = n_samples / n)
+```
+
+```{r fig.height=4, fig.width=12, message = FALSE}
+ggplot(samples_per_celltype_df) + 
+  aes(x = epic_celltype, y = prop_samples) + 
+  geom_col() +
+  geom_hline(yintercept = 0.5, color = "firebrick2") +
+  facet_wrap(vars(project_diagnosis), nrow = 1) +
+  # add guiding line at 0.5
+  theme(axis.text.x = element_text(angle = 30, hjust = 1, size = rel(0.8))) + 
+  ggtitle("Proportion of samples with each cell type")
+```
+
+* `NKcells` are not common in `SCPCP000002` or `SCPCP000009` and generally appear in the fewest samples
+* Most samples contain `CD8` T cells, and `CD4` cells are also common but a bit less than `CD8`
+* Except for `SCPCP000017`, most samples contain `BCells`
+
+### Comparison with mRNAProportions
+
+
+`EPIC` also provides an estimate for cell fractions without correcting for expected mRNA content among cell types; this setting is recommended for pseudobulk samples but not true bulk RNA-seq, so we didn't use it above.
+
+Just to cover all bases though, let's do a quick comparison to see how different the values are. 
+In the plot `corrected_fractions` corresponds to what we have been using; `uncorrected_fractions` corresponds to fractions which have not been corrected for mRNA content, i.e. the `mRNAProportions` quantity.
+
+```{r fig.height=10}
+tref_renamed_df <- tref_df |>
+  dplyr::filter(!is_other) |>
+  dplyr::select(project_diagnosis, epic_celltype, corrected_fractions = fraction, uncorrected_fractions = mRNAProportions)
+
+ggplot(tref_renamed_df) +
+  aes(x = corrected_fractions, y = uncorrected_fractions) + 
+  geom_point() + 
+  geom_abline(color = "blue") + 
+  coord_equal() +
+  facet_grid(
+    rows = vars(epic_celltype), 
+    cols = vars(project_diagnosis)
+  )
+```
+Overall values seem highly similar, but we can see that `Macrophages` have been corrected to have slightly lower values.
+A more in-depth comparison here might zoom in to axes and/or explore relative error between measurements, but that is beyond the scope of what we are interested in looking at which is just to peek and get a sense!
+
+## Conclusions
+
+Between the two `EPIC` references explored here, `TRef` seems more suited to our data since it contains some non-leukocyte normal cell types and does a more thorough job of classification.
+That said, it is not clear whether it _should_ do a more thorough job of classification, and it is missing useful-to-have cell types `Monocytes` and `Neutrophils` which are in the `BRef` reference.
```

Comment:
Exactly! Just because it identifies cells doesn't mean it should, looking at you Osteo 👀

Reply:
oh 💯